src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.netcoreapp.cs

```diff
@@ -24,5 +36,166 @@ public static void Create(this DirectoryInfo directoryInfo, DirectorySecurity di
 
             FileSystem.CreateDirectory(directoryInfo.FullName, directorySecurity.GetSecurityDescriptorBinaryForm());
         }
+
+        /// <summary>
+        /// Creates a new file stream, ensuring it is created with the specified properties and security settings.
+        /// </summary>
+        /// <param name="fileInfo">The current instance describing a file that does not exist in disk yet.</param>
+        /// <param name="mode">One of the enumeration values that specifies how the operating system should open a file.</param>
+        /// <param name="rights">One of the enumeration values that defines the access rights to use when creating access and audit rules.</param>
+        /// <param name="share">One of the enumeration values for controlling the kind of access other FileStream objects can have to the same file.</param>
+        /// <param name="bufferSize">The number of bytes buffered for reads and writes to the file.</param>
+        /// <param name="options">One of the enumeration values that describes how to create or overwrite the file.</param>
+        /// <param name="fileSecurity">An object that determines the access control and audit security for the file.</param>
+        /// <returns>A file stream for the newly created file.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="fileInfo" /> or <paramref name="fileSecurity" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="mode" /> or <paramref name="share" /> are out of their legal enum range.
+        ///-or-
+        /// <paramref name="options" /> is not a supported enum value.
+        ///-or-
+        /// <paramref name="bufferSize" /> is not a positive number.</exception>
+        /// <exception cref="UnauthorizedAccessException">Access to the path is denied.</exception>
+        /// <exception cref="ArgumentException">Invalid handle.
+        ///-or-
+        /// Handle does not support asynchronous operations. The parameters to the <see cref="FileStream" /> constructor may need to be changed to indicate that the handle was opened synchronously (that is, it was not opened for overlapped I/O).
```

Comment:
The `FileStream` constructor we call can throw this async related exception if the handle is created synchronously, but we request a FileStream to be created asynchronously. This shouldn't happen because the file handle is created using `options`, and if it contains `FileOptions.Asynchronous`, the handle will be asynchronous, which maps to `FILE_FLAG_OVERLAPPED`.
See: https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilew

Reply:
Yes, I wouldn't include this as it shouldn't happen.